Invalid assertion in LastExecutionTests.lastExecutionGetRunningTimeTest causes infrequent intermittent failure of the test

We are seeing the following TCK test fail intermittently (extremely infrequently - just 3 times since we automated running of the TCK several months ago)

ee.jakarta.tck.concurrent.api.LastExecution.LastExecutionTests.lastExecutionGetRunningTimeTest

which intermittently fails claiming that a repeating task that was scheduled with a custom Trigger only ran once instead of twice: `java.lang.AssertionError: Got wrong last execution result. expected [2] but found [1]`

We can see from our product trace that the task executes the first time, updates the ScheduledFuture according to the first result, then schedules the second execution to run a second or so later.  In the mean time, the test case, which invokes ScheduledFuture.isDone as its way of determining that _both_ executions have completed, sees a true result, checks its own counter, and fails with this error.  The problem is that documentation of `ScheduledFuture.isDone` is absent and inherited from the superclass.  `Future.isDone` makes no mention of behavior when a single Future represents a task that executes multiple times.  The TCK test is wrong to assert that `ScheduledFuture.isDone` is a reliable mechanism for waiting for all executions to be done vs the current execution or other meaning.  This test could be challenged, however, the relative infrequency with which it occurs means it should be fine to just correct it in Jakarta EE 11.  An easy correction would involve waiting on the counter value instead of the ambiguous isDone signal.

```
java.lang.AssertionError: Got wrong last execution result. expected [2] but found [1]
at org.testng.Assert.fail(Assert.java:96)
at org.testng.Assert.failNotEquals(Assert.java:776)
at org.testng.Assert.assertEqualsImpl(Assert.java:137)
at org.testng.Assert.assertEquals(Assert.java:118)
at org.testng.Assert.assertEquals(Assert.java:652)
at ee.jakarta.tck.concurrent.framework.ArquillianTests.assertEquals(ArquillianTests.java:120)
at ee.jakarta.tck.concurrent.api.LastExecution.LastExecutionTests.lastExecutionGetRunningTimeTest(LastExecutionTests.java:131)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
at org.testng.internal.MethodInvocationHelper$1.runTestMethod(MethodInvocationHelper.java:230)
at org.jboss.arquillian.testng.Arquillian$3.invoke(Arquillian.java:146)
at org.jboss.arquillian.container.test.impl.execution.LocalTestExecuter.execute(LocalTestExecuter.java:57)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
at org.jboss.arquillian.core.impl.EventImpl.fire(EventImpl.java:62)
at org.jboss.arquillian.container.test.impl.execution.ContainerTestExecuter.execute(ContainerTestExecuter.java:36)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
at org.jboss.arquillian.test.impl.TestContextHandler.createTestContext(TestContextHandler.java:116)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
at org.jboss.arquillian.test.impl.TestContextHandler.createClassContext(TestContextHandler.java:83)
at jdk.internal.reflect.GeneratedMethodAccessor51.invoke(Unknown Source)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
at org.jboss.arquillian.test.impl.TestContextHandler.createSuiteContext(TestContextHandler.java:69)
at jdk.internal.reflect.GeneratedMethodAccessor50.invoke(Unknown Source)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
at org.jboss.arquillian.test.impl.EventTestRunnerAdaptor.test(EventTestRunnerAdaptor.java:139)
at org.jboss.arquillian.testng.Arquillian.run(Arquillian.java:138)
at org.testng.internal.MethodInvocationHelper.invokeHookable(MethodInvocationHelper.java:242)
at org.testng.internal.Invoker.invokeMethod(Invoker.java:579)
at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
at org.testng.TestRunner.privateRun(TestRunner.java:648)
at org.testng.TestRunner.run(TestRunner.java:505)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
at org.testng.SuiteRunner.run(SuiteRunner.java:364)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
at org.testng.TestNG.runSuites(TestNG.java:1049)
at org.testng.TestNG.run(TestNG.java:1017)
at org.jboss.arquillian.testng.container.TestNGTestRunner.execute(TestNGTestRunner.java:51)
at org.jboss.arquillian.protocol.servlet5.runner.ServletTestRunner.executeTest(ServletTestRunner.java:139)
at org.jboss.arquillian.protocol.servlet5.runner.ServletTestRunner.execute(ServletTestRunner.java:117)
at org.jboss.arquillian.protocol.servlet5.runner.ServletTestRunner.doGet(ServletTestRunner.java:86)
at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:500)
at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:587)
at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1258)
at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:746)
at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:443)
at com.ibm.ws.webcontainer.filter.WebAppFilterChain.invokeTarget(WebAppFilterChain.java:193)
at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:98)
at com.ibm.ws.security.jaspi.JaspiServletFilter.doFilter(JaspiServletFilter.java:56)
at com.ibm.ws.webcontainer.filter.FilterInstanceWrapper.doFilter(FilterInstanceWrapper.java:201)
at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:91)
at com.ibm.ws.webcontainer.filter.WebAppFilterManager.doFilter(WebAppFilterManager.java:1002)
at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1140)
at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1011)
at com.ibm.ws.webcontainer.servlet.CacheServletWrapper.handleRequest(CacheServletWrapper.java:75)
at com.ibm.ws.webcontainer40.servlet.CacheServletWrapper40.handleRequest(CacheServletWrapper40.java:85)
at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:938)
at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.run(DynamicVirtualHost.java:281)
at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink$TaskWrapper.run(HttpDispatcherLink.java:1184)
at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.wrapHandlerAndExecute(HttpDispatcherLink.java:453)
at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.ready(HttpDispatcherLink.java:412)
at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:566)
at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleNewRequest(HttpInboundLink.java:500)
at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.processRequest(HttpInboundLink.java:360)
at com.ibm.ws.http.channel.internal.inbound.HttpICLReadCallback.complete(HttpICLReadCallback.java:70)
at com.ibm.ws.tcpchannel.internal.WorkQueueManager.requestComplete(WorkQueueManager.java:514)
at com.ibm.ws.tcpchannel.internal.WorkQueueManager.attemptIO(WorkQueueManager.java:584)
at com.ibm.ws.tcpchannel.internal.WorkQueueManager.workerRun(WorkQueueManager.java:968)
at com.ibm.ws.tcpchannel.internal.WorkQueueManager$Worker.run(WorkQueueManager.java:1057)
at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:245)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
at java.base/java.lang.Thread.run(Thread.java:866)
```

--- a/tck/src/main/java/ee/jakarta/tck/concurrent/api/LastExecution/LastExecutionTests.java
+++ b/tck/src/main/java/ee/jakarta/tck/concurrent/api/LastExecution/LastExecutionTests.java
@@ -16,11 +16,8 @@
 
 package ee.jakarta.tck.concurrent.api.LastExecution;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
-
 import java.util.HashMap;
 import java.util.Map;
-import java.util.concurrent.ScheduledFuture;
 
 import org.jboss.arquillian.container.test.api.Deployment;
 import org.jboss.shrinkwrap.api.ShrinkWrap;
@@ -36,7 +33,6 @@ import ee.jakarta.tck.concurrent.framework.junit.anno.Common;
 import ee.jakarta.tck.concurrent.framework.junit.anno.Common.PACKAGE;
 import ee.jakarta.tck.concurrent.framework.junit.anno.TestName;
 import ee.jakarta.tck.concurrent.framework.junit.anno.Web;
-import ee.jakarta.tck.concurrent.framework.junit.extensions.Wait;
 import jakarta.annotation.Resource;
 import jakarta.enterprise.concurrent.ManagedExecutors;
 import jakarta.enterprise.concurrent.ManagedScheduledExecutorService;
@@ -72,51 +68,40 @@ public class LastExecutionTests {
         Map<String, String> executionProperties = new HashMap<String, String>();
         executionProperties.put(ManagedTask.IDENTITY_NAME, IDENTITY_NAME_TEST_ID);
 
-        ScheduledFuture<?> sf = scheduledExecutor.schedule(
+        scheduledExecutor.schedule(
                 ManagedExecutors.managedTask(new CounterRunnableTask(), executionProperties, null),
                 new LogicDrivenTrigger(TestConstants.pollInterval.toMillis(), testname));
-        Wait.waitTillFutureIsDone(sf);
-
-        assertEquals(LogicDrivenTrigger.RIGHT_COUNT, // expected
-                StaticCounter.getCount(), // actual
-                "Got wrong identity name. See server log for more details.");
+        
+        StaticCounter.waitTill(LogicDrivenTrigger.RIGHT_COUNT, "Got wrong identity name. See server log for more details.");
     }
 
     @Assertion(id = "JAVADOC:16", strategy = "Result of the last execution.")
     public void lastExecutionGetResultRunnableTest() {
         // test with runnable, LastExecution should return null
-        ScheduledFuture<?> sf = scheduledExecutor.schedule(
+        scheduledExecutor.schedule(
                 ManagedExecutors.managedTask(new CounterRunnableTask(), null, null),
                 new LogicDrivenTrigger(TestConstants.pollInterval.toMillis(), testname));
-        Wait.waitTillFutureIsDone(sf);
-
-        assertEquals(LogicDrivenTrigger.RIGHT_COUNT, // expected
-                StaticCounter.getCount(), // actual
-                "Got wrong last execution result. See server log for more details.");
+        
+        StaticCounter.waitTill(LogicDrivenTrigger.RIGHT_COUNT, "Got wrong last execution result. See server log for more details.");
     }
 
     @Assertion(id = "JAVADOC:16", strategy = "Result of the last execution.")
     public void lastExecutionGetResultCallableTest() {
         // test with callable, LastExecution should return 1
-        ScheduledFuture<?> sf = scheduledExecutor.schedule(
+        scheduledExecutor.schedule(
                 ManagedExecutors.managedTask(new CounterCallableTask(), null, null),
                 new LogicDrivenTrigger(TestConstants.pollInterval.toMillis(), testname));
-        Wait.waitTillFutureIsDone(sf);
-
-        assertEquals(LogicDrivenTrigger.RIGHT_COUNT, // expected
-                StaticCounter.getCount(), // actual
-                "Got wrong last execution result. See server log for more details.");
+        
+        StaticCounter.waitTill(LogicDrivenTrigger.RIGHT_COUNT, "Got wrong last execution result. See server log for more details.");
     }
 
     @Assertion(id = "JAVADOC:17 JAVADOC:18 JAVADOC:19", strategy = "The last time in which the task was completed.")
     public void lastExecutionGetRunningTimeTest() {
-        ScheduledFuture<?> sf = scheduledExecutor.schedule(
+        scheduledExecutor.schedule(
                 ManagedExecutors.managedTask(new CounterRunnableTask(TestConstants.pollInterval), null, null),
                 new LogicDrivenTrigger(TestConstants.pollInterval.toMillis(), testname));
-        Wait.waitTillFutureIsDone(sf);
-        assertEquals(LogicDrivenTrigger.RIGHT_COUNT, // expected
-                StaticCounter.getCount(), // actual
-                "Got wrong last execution result.");
+        
+        StaticCounter.waitTill(LogicDrivenTrigger.RIGHT_COUNT, "Got wrong last execution result. See server log for more details.");
     }
 
 }

--- a/tck/src/main/java/ee/jakarta/tck/concurrent/api/Trigger/TriggerTests.java
+++ b/tck/src/main/java/ee/jakarta/tck/concurrent/api/Trigger/TriggerTests.java
@@ -72,7 +72,7 @@ public class TriggerTests {
             Assertions.assertBetween(StaticCounter.getCount(), TestConstants.pollsPerTimeout - 2,
                     TestConstants.pollsPerTimeout + 2);
         } finally {
-            Wait.waitTillFutureIsDone(result);
+            Wait.waitForTaskComplete(result);
         }
     }
 

--- a/tck/src/main/java/ee/jakarta/tck/concurrent/common/fixed/counter/StaticCounter.java
+++ b/tck/src/main/java/ee/jakarta/tck/concurrent/common/fixed/counter/StaticCounter.java
@@ -44,11 +44,15 @@ public final class StaticCounter {
     }
 
     public static void waitTill(final int expected) {
+        waitTill(expected, "Expected count " + expected + " within timeout.");
+    }
+    
+    public static void waitTill(final int expected, final String message) {
         assertTimeoutPreemptively(TestConstants.waitTimeout, () -> {
             for (; expected != StaticCounter.getCount(); Wait.sleep(TestConstants.pollInterval)) {
                 //empty
             }
-        });
+        }, message);
     }
 
     public static void waitTillSurpassed(final int expected) {

--- a/tck/src/main/java/ee/jakarta/tck/concurrent/framework/junit/extensions/Wait.java
+++ b/tck/src/main/java/ee/jakarta/tck/concurrent/framework/junit/extensions/Wait.java
@@ -73,25 +73,6 @@ public final class Wait {
     }
 
     /**
-     * Waits for future to complete, but will timeout after
-     * {@link TestConstants#waitTimeout}, and will be polled every
-     * {@link TestConstants#pollInterval}
-     *
-     * The difference between this method and waitForTaskComplete is that some
-     * scheduled task will return values for multiple times, in this situation
-     * waitForTaskComplete does not work.
-     *
-     * @param future - the future to wait for
-     */
-    public static void waitTillFutureIsDone(final Future<?> future) {
-        assertTimeoutPreemptively(TestConstants.waitTimeout, () -> {
-            for (; !future.isDone(); sleep(TestConstants.pollInterval)) {
-                //empty
-            }
-        });
-    }
-
-    /**
      * Waits for future to throw an error, but will timeout after
      * {@link TestConstants#waitTimeout}, and will be polled every
      * {@link TestConstants#pollInterval}

--- a/tck/src/main/java/ee/jakarta/tck/concurrent/spec/ManagedExecutorService/inheritedapi/InheritedAPITests.java
+++ b/tck/src/main/java/ee/jakarta/tck/concurrent/spec/ManagedExecutorService/inheritedapi/InheritedAPITests.java
@@ -72,16 +72,13 @@ public class InheritedAPITests {
             strategy = "Test basic function for ManagedExecutorService: submit")
     public void testSubmit() throws Exception {
         Future<?> result = executor.submit(new CommonTasks.SimpleCallable());
-        Wait.waitTillFutureIsDone(result);
-        assertEquals(result.get(), TestConstants.simpleReturnValue);
+        assertEquals(Wait.waitForTaskComplete(result), TestConstants.simpleReturnValue);
 
         result = executor.submit(new CommonTasks.SimpleRunnable());
-        Wait.waitTillFutureIsDone(result);
-        result.get();
+        Wait.waitForTaskComplete(result);
 
         result = executor.submit(new CommonTasks.SimpleRunnable(), TestConstants.simpleReturnValue);
-        Wait.waitTillFutureIsDone(result);
-        assertEquals(result.get(), TestConstants.simpleReturnValue);
+        assertEquals(Wait.waitForTaskComplete(result), TestConstants.simpleReturnValue);
     }
 
     @Assertion(id = "SPEC:14.4; SPEC:6.1; SPEC:6.2; SPEC:8",
@@ -119,7 +116,7 @@ public class InheritedAPITests {
             taskList.add(new CommonTasks.SimpleArgCallable(3));
             List<Future<Integer>> resultList = executor.invokeAll(taskList);
             for (Future<?> each : resultList) {
-                Wait.waitTillFutureIsDone(each);
+                Wait.waitForTaskComplete(each);
             }
             assertEquals(resultList.get(0).get(), 1);
             assertEquals(resultList.get(1).get(), 2);
@@ -127,7 +124,7 @@ public class InheritedAPITests {
 
             resultList = executor.invokeAll(taskList, TestConstants.waitTimeout.getSeconds(), TimeUnit.SECONDS);
             for (Future<?> each : resultList) {
-                Wait.waitTillFutureIsDone(each);
+                Wait.waitForTaskComplete(each);
             }
             assertEquals(resultList.get(0).get(), 1);
             assertEquals(resultList.get(1).get(), 2);

--- a/tck/src/main/java/ee/jakarta/tck/concurrent/spec/ManagedScheduledExecutorService/inheritedapi/InheritedAPIServletTests.java
+++ b/tck/src/main/java/ee/jakarta/tck/concurrent/spec/ManagedScheduledExecutorService/inheritedapi/InheritedAPIServletTests.java
@@ -67,16 +67,13 @@ public class InheritedAPIServletTests {
     @Assertion(id = "SPEC:44.1", strategy = "Test basic function for ManagedScheduledExecutorService: submit")
     public void testApiSubmit() throws Exception {
         Future<?> result = scheduledExecutor.submit(new CommonTasks.SimpleCallable());
-        Wait.waitTillFutureIsDone(result);
-        assertEquals(result.get(), TestConstants.simpleReturnValue);
+        assertEquals(Wait.waitForTaskComplete(result), TestConstants.simpleReturnValue);
 
         result = scheduledExecutor.submit(new CommonTasks.SimpleRunnable());
-        Wait.waitTillFutureIsDone(result);
-        result.get();
+        Wait.waitForTaskComplete(result);
 
         result = scheduledExecutor.submit(new CommonTasks.SimpleRunnable(), TestConstants.simpleReturnValue);
-        Wait.waitTillFutureIsDone(result);
-        assertEquals(result.get(), TestConstants.simpleReturnValue);
+        assertEquals(Wait.waitForTaskComplete(result), TestConstants.simpleReturnValue);
     }
 
     @Assertion(id = "SPEC:44.2", strategy = "Test basic function for ManagedScheduledExecutorService: execute")
@@ -98,7 +95,7 @@ public class InheritedAPIServletTests {
             taskList.add(new CommonTasks.SimpleArgCallable(3));
             List<Future<Integer>> resultList = scheduledExecutor.invokeAll(taskList);
             for (Future<?> each : resultList) {
-                Wait.waitTillFutureIsDone(each);
+                Wait.waitForTaskComplete(each);
             }
             assertEquals(resultList.get(0).get(), 1);
             assertEquals(resultList.get(1).get(), 2);
@@ -106,7 +103,7 @@ public class InheritedAPIServletTests {
             resultList = scheduledExecutor.invokeAll(taskList, TestConstants.waitTimeout.getSeconds(),
                     TimeUnit.SECONDS);
             for (Future<?> each : resultList) {
-                Wait.waitTillFutureIsDone(each);
+                Wait.waitForTaskComplete(each);
             }
             assertEquals(resultList.get(0).get(), 1);
             assertEquals(resultList.get(1).get(), 2);
@@ -154,13 +151,11 @@ public class InheritedAPIServletTests {
     public void testApiSchedule() throws Exception {
         Future<?> result = scheduledExecutor.schedule(new CommonTasks.SimpleCallable(),
                 TestConstants.pollInterval.getSeconds(), TimeUnit.SECONDS);
-        Wait.waitTillFutureIsDone(result);
-        assertEquals(result.get(), TestConstants.simpleReturnValue);
+        assertEquals(Wait.waitForTaskComplete(result), TestConstants.simpleReturnValue);
 
         result = scheduledExecutor.schedule(new CommonTasks.SimpleRunnable(), TestConstants.pollInterval.getSeconds(),
                 TimeUnit.SECONDS);
-        Wait.waitTillFutureIsDone(result);
-        assertEquals(result.get(), null);
+        assertEquals(Wait.waitForTaskComplete(result), null);
     }
 
     @Assertion(id = "SPEC:44.6", strategy = "Test basic function for ManagedScheduledExecutorService: scheduleAtFixedRate")

--- a/tck/src/main/java/ee/jakarta/tck/concurrent/spec/ManagedScheduledExecutorService/inheritedapi/TestEjb.java
+++ b/tck/src/main/java/ee/jakarta/tck/concurrent/spec/ManagedScheduledExecutorService/inheritedapi/TestEjb.java
@@ -54,16 +54,13 @@ public class TestEjb implements TestEjbInterface {
     public void testApiSubmit() {
         try {
             Future<?> result = scheduledExecutor.submit(new CommonTasks.SimpleCallable());
-            Wait.waitTillFutureIsDone(result);
-            assertEquals(result.get(), TestConstants.simpleReturnValue);
+            assertEquals(Wait.waitForTaskComplete(result), TestConstants.simpleReturnValue);
 
             result = scheduledExecutor.submit(new CommonTasks.SimpleRunnable());
-            Wait.waitTillFutureIsDone(result);
-            result.get();
+            Wait.waitForTaskComplete(result);
 
             result = scheduledExecutor.submit(new CommonTasks.SimpleRunnable(), TestConstants.simpleReturnValue);
-            Wait.waitTillFutureIsDone(result);
-            assertEquals(result.get(), TestConstants.simpleReturnValue);
+            assertEquals(Wait.waitForTaskComplete(result), TestConstants.simpleReturnValue);
         } catch (Exception e) {
             throw new RuntimeException(e);
         }
@@ -87,7 +84,7 @@ public class TestEjb implements TestEjbInterface {
             taskList.add(new CommonTasks.SimpleArgCallable(3));
             List<Future<Integer>> resultList = scheduledExecutor.invokeAll(taskList);
             for (Future<?> each : resultList) {
-                Wait.waitTillFutureIsDone(each);
+                Wait.waitForTaskComplete(each);
             }
             assertEquals(resultList.get(0).get(), 1);
             assertEquals(resultList.get(1).get(), 2);
@@ -96,7 +93,7 @@ public class TestEjb implements TestEjbInterface {
             resultList = scheduledExecutor.invokeAll(taskList, TestConstants.waitTimeout.getSeconds(),
                     TimeUnit.SECONDS);
             for (Future<?> each : resultList) {
-                Wait.waitTillFutureIsDone(each);
+                Wait.waitForTaskComplete(each);
             }
             assertEquals(resultList.get(0).get(), 1);
             assertEquals(resultList.get(1).get(), 2);
@@ -151,13 +148,11 @@ public class TestEjb implements TestEjbInterface {
         try {
             Future<?> result = scheduledExecutor.schedule(new CommonTasks.SimpleCallable(),
                     TestConstants.pollInterval.getSeconds(), TimeUnit.SECONDS);
-            Wait.waitTillFutureIsDone(result);
-            assertEquals(TestConstants.simpleReturnValue, result.get());
+            assertEquals(TestConstants.simpleReturnValue, Wait.waitForTaskComplete(result));
 
             result = scheduledExecutor.schedule(new CommonTasks.SimpleRunnable(),
                     TestConstants.pollInterval.getSeconds(), TimeUnit.SECONDS);
-            Wait.waitTillFutureIsDone(result);
-            assertEquals(result.get(), null);
+            assertEquals(null, Wait.waitForTaskComplete(result));
         } catch (Exception e) {
             fail(e.getMessage());
         }

